### ISSUE TITLE
Does not compile with OTP 23 (Mac OS)

### ISSUE THREAD

**Issue by zookzook** [elixir-cldr/cldr_collation#2]
Hey! 

using OTP 23 I got the linking error while compiling my project (mix compile):

```
cc c_src/ucol.o -arch x86_64 -flat_namespace -undefined suppress -shared -L/usr/local/Cellar/erlang/23.2.2/lib/erlang/usr/lib -lerl_interface -lei -lpthread -lm -licucore -lstdc++ -o ./priv/ucol.so
ld: library not found for -lerl_interface
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [priv/ucol.so] Error 1
```

It looks that the lib `erl_interface` was removed (maybe renamed) starting with OTP 23. The lib folder of OTP 23 contains

```
liberts.a
libei_st.a
libei.a
liberts_r.a
```

while OTP 22 the lib folder contains

```
liberl_interface.a
libei_st.a
libei.a
liberl_interface_st.a
liberts_r.a
liberts.a
```

**Comment by kipcole9** [comment]
Thanks for the report and apologies for the inconvenience.  Looks like `liberl_interface.a` is now obsolete and `libie.a` is all that is required.

I will work on a new release but for now if you update the `Makefile` thats in the library and change:
```bash
LDLIBS += -L$(ERL_INTERFACE_LIB_DIR) -lerl_interface -lei
```
to
```bash
LDLIBS += -L$(ERL_INTERFACE_LIB_DIR) -lei
```
Then it compiles and the tests pass. I'll work on this over the weekend - does that timing work ok for you?

**Comment by kipcole9** [comment]
As of [this commit](https://github.com/elixir-cldr/cldr_collation/commit/51b20f401e9de95dd303d039cf69b778ec2af343) I think I am correctly conditionally linking based upon OTP version (actually, based upon whether `liberl_interface.a` exists or not.

Would you mind trying to use the version from master to see if it works for you? I need to do some more testing on various OTP releases before publishing a new version which I will do tomorrow.

To update in `mix.exs`:
```elixir
def deps() do
  [
    {:ex_cldr_collation, github: "elixir-cldr/cldr_collation"},
    ...
  ]
end
```
Then:
```elixir
mix compile
```

Let me know if that works ok?

**Comment by zookzook** [comment]
Thank you very much! I will check it now! Thank you for your fast response!

**Comment by zookzook** [comment]
Yes, it works! Again: thank you very much! Never seen such a fast response before!

**Comment by kipcole9** [comment]
Happy to help. Disappointed I hadn't tested it earlier too. But appreciate the feedback - it helps motivation for keeping this "weekend project" going!  Will keep the issue open until I have published a new version after testing on earlier OTP releases.

**Comment by kipcole9** [comment]
I had a bit of time so I installed and tested on OTP 22, OTP 23 and OTP 24 release candidate, all with success. So I have published [ex_cldr_collation version 0.4.0](https://hex.pm/packages/ex_cldr_collation/0.4.0).  The changelog entry is:

### Bug Fixes

* Fix compilation issues on OTP 23 and later. On these releases, `liberl_interface.a` doesn't exist and isn't required.  Thanks to @zookzook for the report.

**Comment by zookzook** [comment]
Thank you very much! I changed mix and updated the version, everything is running. Fasted bug fix ever!
